### ISSUE TITLE
Bug: [no-deprecated] doesn't report deprecated type index access

### ISSUE THREAD

**Issue by ronami** [typescript-eslint/typescript-eslint#10837]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=C4TwDgpgBAKhDOwoF4oG8BQUoHoBUeUAAgCYRgBOEAxgIbAQlR45ZS0BcUA5LdwDRsARl25CBGAL4BuDBhw4o8ABYB7AK4AbJhAoVVFKEPVISqhADtuwDKEhQASgi1JUcRAG1e3ALqygA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dGADakRNGgdokALrgwAXxBqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
type Test = {
  /** @deprecated */
  a: 'a',
  b: 'b',
};

// should error but doesn't
type Result = Test['a'];
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-deprecated": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected the deprecated type index access to be reported since it's also reported by TypeScript as deprecated.

### Actual Result

The deprecated type index access hasn't been reported.

**Comment by ronami** [comment]
Interestingly, this seems to also affect non-type (computed) index access ([playground link](https://typescript-eslint.io/play/#ts=5.3.3&fileType=.ts&code=MYewdgzgLgBFCm0CMMC8MDeAoGMD0AVATAAIAm8ADgE7zACGCZMBeOMA5AGYggC0AI3rUOALk70OAGiwBfANxYsePDAgALEAFcANs1qUQ1WAK2wyIRGA5QsoSLFoRdUFOgTIA2t16DhHAF1FO3BoOEQoACY0THZCYnIqWgYmFjZcTwBlAE8AWwEQHQA6AEsEakYjAPEkGQUlFTVNXX0qIxMzGAsrGxCHGCcXaPcIyKy8guKy%2BAqoKsUgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oBNFjpEZAIb5E3dGADakRNGgdokALrgwAXxBqgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)):

```ts
const test1 = {
  /** @deprecated */
  'foo-bar': 'a',
};

// should report but doesn't
const result1 = test1['foo-bar'];

const test2 = {
  /** @deprecated */
  [Symbol.iterator]: 1,
};

// should report but doesn't
const result2 = test2[Symbol.iterator];
```
